File: GolPredictor.Web/src/app/core/models/partido-dto.ts
import { PaisDto } from "./pais-dto";

export interface PartidoDto {
  id: number;
  team1Id: number | null;
  team2Id: number | null;
  fechaInicio: Date | null;
  fechaFin: Date | null;
  status: boolean | null;
  resultTeam1: number | null;
  resultTeam2: number | null;
  team1: PaisDto;
  team2: PaisDto;
}

File: GolPredictor.WebApi/GolPredictor.WebApi/DTO/Partido/PartidoDto.cs
using GolPredictor.WebApi.DTO.Pais;
using System;

namespace GolPredictor.WebApi.DTO.Partido
{
    public class PartidoDto
    {
        public int Id { get; set; }
        public int? Team1Id { get; set; }
        public int? Team2Id { get; set; }
        public DateTime? FechaInicio { get; set; }
        public DateTime? FechaFin { get; set; }
        public bool? Status { get; set; }
        public int? ResultTeam1 { get; set; }
        public int? ResultTeam2 { get; set; }
        public virtual PaisDto Team1 { get; set; }
        public virtual PaisDto Team2 { get; set; }
    }
}
